[Bug report] GravitinoVirtualFileSystem.getScheme() throws UnsupportedOperationException, breaking Spark/MapReduce commit jobs

### Version

main branch

### Describe what's wrong

`GravitinoVirtualFileSystem` extends `org.apache.hadoop.fs.FileSystem` but does not override `getScheme()`. The base implementation throws `UnsupportedOperationException`, so any caller that invokes `fs.getScheme()` on a `gvfs://` filesystem fails.

It should simply return the GVFS scheme (`gvfs`, already defined as `GravitinoVirtualFileSystemConfiguration.GVFS_SCHEME`), consistent with other Hadoop `FileSystem` implementations (HDFS → `hdfs`, S3A → `s3a`, etc.).

### Error message and/or stacktrace

```
Caused by: java.lang.UnsupportedOperationException:
    Not implemented by the GravitinoVirtualFileSystem FileSystem implementation
        at org.apache.hadoop.fs.FileSystem.getScheme(FileSystem.java:305)
        at org.apache.hadoop.mapreduce.lib.output.FileOutputCommitter.lambda$commitJobInternal$0(...)
```

### How to reproduce

```java
Configuration conf = new Configuration();
// ... GVFS configuration ...
FileSystem fs = FileSystem.get(URI.create("gvfs://fileset/metalake/catalog/schema/fs1"), conf);
fs.getScheme(); // throws UnsupportedOperationException
```

### Additional context

_No response_

--- a/clients/filesystem-hadoop3/src/main/java/org/apache/gravitino/filesystem/hadoop/GravitinoVirtualFileSystem.java
+++ b/clients/filesystem-hadoop3/src/main/java/org/apache/gravitino/filesystem/hadoop/GravitinoVirtualFileSystem.java
@@ -123,6 +123,12 @@ public class GravitinoVirtualFileSystem extends FileSystem {
     return this.uri;
   }
 
+  /** Returns the GVFS scheme to avoid {@link UnsupportedOperationException} in commit jobs. */
+  @Override
+  public String getScheme() {
+    return GravitinoVirtualFileSystemConfiguration.GVFS_SCHEME;
+  }
+
   @Override
   public synchronized Path getWorkingDirectory() {
     return this.workingDirectory;

--- a/clients/filesystem-hadoop3/src/test/java/org/apache/gravitino/filesystem/hadoop/TestGvfsBase.java
+++ b/clients/filesystem-hadoop3/src/test/java/org/apache/gravitino/filesystem/hadoop/TestGvfsBase.java
@@ -257,6 +257,17 @@ public class TestGvfsBase extends GravitinoMockServerBase {
   }
 
   @Test
+  public void testGetScheme() throws IOException {
+    Assumptions.assumeTrue(getClass() == TestGvfsBase.class);
+    try (FileSystem fs = new Path("gvfs://fileset/").getFileSystem(conf)) {
+      // The default FileSystem#getScheme() throws UnsupportedOperationException, which breaks
+      // Hadoop / Spark commit protocols when writing to a gvfs:// path. GVFS must override it
+      // and return the GVFS scheme constant.
+      assertEquals(GravitinoVirtualFileSystemConfiguration.GVFS_SCHEME, fs.getScheme());
+    }
+  }
+
+  @Test
   public void testRequestHeaders()
       throws NoSuchFieldException, IllegalAccessException, IOException {
     String envKey = "GRAVITINO_TEST_HEADER";

